src/lib/common/string.cpp

```diff
@@ -345,6 +405,21 @@ bool parseUrl(const std::string& url, std::string& host, int& port, std::string&
     }
   }
 
+  //
+  // Is 'port' a valid number?
+  // Remember the port is really an 'unsigned short', so numbers over MAXSHORT (0xFFFF) are
+  // not valid. Negative port numbers cannot exist and 0 is reserved.
+  //
+  if ((port > MAXSHORT) || (port <= 0))
```

Comment:
Valid port are a range betweeh 0 and 65365 (by desing of the TCP/IP stack). In our case it happens that MAXSHORT matches 65365 but it is an "implementation side effect". I think that using MAX_PORT (defined in limits.h) would be more explicit.